.github/settings.yml

```diff
@@ -0,0 +1,33 @@
+repository:
+  delete_branch_on_merge: true
+  has_wiki: false
+  has_projects: false
+  allow_squash_merge: true
+  allow_merge_commit: false
+  allow_rebase_merge: false
+
+branches:
+  - name: master
+    protection:
+      required_pull_request_reviews:
+        required_approving_review_count: 1
+        dismiss_stale_reviews: false
+        require_code_owner_reviews: true
+        dismissal_restrictions: null
+      required_status_checks:
+        strict: false
+        contexts: ['test', 'WIP']
+  - name: main
+    protection:
+      required_pull_request_reviews:
+        required_approving_review_count: 1
+        dismiss_stale_reviews: false
+        require_code_owner_reviews: true
+        dismissal_restrictions: null
+      required_status_checks:
+        strict: false
+        contexts: ['test', 'WIP']
```

Comment:
should we also add branch protection for the pre- and maintenance releases?

- `next`
- `beta`
- `*.x`

Reply:
unfortunately, i believe the api is limited from setting up protection on branches that dont exist. i just confirmed with https://github.com/travi-test/settings-test/commit/17cba6cf2eb6578547f813cce6e615a65f9d6bd8 that its not working. if you happen to have insight to suggest this _should_ work, I would love to be able to support this.

do we currently define this type of protection? the other thing to consider, if so is that it might get removed if we dont define it

Reply:
dug a little deeper to at least confirm that the current issue is a 404 result with message "Branch not found" for 

```
method: "PUT"
url: "https://api.github.com/repos/travi-test/settings-test/branches/alpha/protection"
```

Reply:
it looks like we at least wont remove existing protection rules when not defined in the file

Reply:
good to know, I thought I had branch protection for all release branches implemented before, but [I only created the issue as a reminder](https://github.com/octokit/create-octokit-project.js/issues/14). Thanks for looking into this.

All good then, let's ship!

Reply:
looking back through settings issues, i did find this mention that it should work: https://github.com/probot/settings/issues/136#issuecomment-634460085. my tests, even last night, through octokit rest suggest otherwise, but i havent dug much deeper than our current implementation and the octokit rest docs. 

i dont find a mention of `createBranchProtectionRule` directly in the octokit docs, so i would have to follow up to better understand. are you familiar with an approach other than [what is currently happening](https://github.com/probot/settings/blob/9d2551e2b7e314f55f28871c7e67a612716239cf/lib/plugins/branches.js#L21) to adjust with octokit rest? i'm also open to going down the graphql route, but not sure where to start without some digging.